src/parser/csql_grammar.y

```diff
@@ -5341,31 +5395,67 @@ meta_class_spec
 			  {
 			    ocs->info.spec.entity_name = $2;
 			    ocs->info.spec.only_all = PT_ONLY;
-			    ocs->info.spec.meta_class = PT_CLASS;
+			    ocs->info.spec.meta_class = PT_META_CLASS;
 			  }
 
-			if (ocs)
-			  ocs->info.spec.meta_class = PT_META_CLASS;
 			$$ = ocs;
 			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
 
 		DBG_PRINT}}
-	;
+        | CLASS class_name_with_server_name
+                {{DBG_TRACE_GRAMMAR(meta_class_spec, : CLASS class_name_with_server_name );
+                       PT_NODE *ocs = parser_new_node (this_parser, PT_SPEC);
+			if (ocs)
+			  {
+                            ocs->info.spec.entity_name = CONTAINER_AT_0 ($2);
+			    /* for DBLink DML */
+                            ocs->info.spec.remote_server_name = CONTAINER_AT_1 ($2);
+			    ocs->info.spec.only_all = PT_ONLY;
+                            ocs->info.spec.meta_class = PT_META_CLASS;
+			  }
 
-only_all_class_spec
-	: only_class_name opt_partition_spec
-		{{ DBG_TRACE_GRAMMAR(only_all_class_spec, : only_class_name opt_partition_spec );
+                        $$ = ocs;
+                DBG_PRINT}}
+	;
 
-			PT_NODE *ocs = parser_new_node (this_parser, PT_SPEC);
+only_all_class_spec_with_server
+	: class_name_with_server_name opt_partition_spec
+		{{ DBG_TRACE_GRAMMAR(only_all_class_spec_with_server, : class_name_with_server_name opt_partition_spec );
+                       PT_NODE *ocs = parser_new_node (this_parser, PT_SPEC);
 			if (ocs)
 			  {
-			    ocs->info.spec.entity_name = $1;
+                            ocs->info.spec.entity_name = CONTAINER_AT_0 ($1);
+			    /* for DBLink DML */
+                            ocs->info.spec.remote_server_name = CONTAINER_AT_1 ($1);
+
 			    ocs->info.spec.only_all = PT_ONLY;
 			    ocs->info.spec.meta_class = PT_CLASS;
 			    if ($2)
-			      {
-				ocs->info.spec.partition = $2;
+		   	      {
+			        ocs->info.spec.partition = $2;
 			      }
+			   }
+
+			$$ = ocs;
+			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
+		DBG_PRINT}}
+	;
+
+only_all_class_spec
```

Comment:
Can't we just add the following rule to only_all_class_spec instead of creating a new only_all_class_spec_with_server rule?
```
only_all_class_spec
	: only_class_name opt_partition_spec
        | only_class_name '@' dblink_server_name opt_partition_spec /* add */
        ...
```
This avoids using container_2 and simplifies the rules.